File: i am tired/blue/1.cpp
#include <iostream>
#include <vector>
using namespace std;

const int N = 5; // 棋盘大小
int drawCount = 0; // 平局计数

// 检查是否有五子连珠
bool hasFiveInARow(vector<vector<char>>& board, char player) {
    // 检查横向和纵向
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            if (j <= N - 5) { // 横向
                bool win = true;
                for (int k = 0; k < 5; ++k)
                    if (board[i][j + k] != player) {
                        win = false;
                        break;
                    }
                if (win) return true;
            }
            if (i <= N - 5) { // 纵向
                bool win = true;
                for (int k = 0; k < 5; ++k)
                    if (board[i + k][j] != player) {
                        win = false;
                        break;
                    }
                if (win) return true;
            }
        }
    }
    // 检查斜线
    for (int i = 0; i <= N - 5; ++i) {
        for (int j = 0; j <= N - 5; ++j) {
            // 右下斜
            bool win = true;
            for (int k = 0; k < 5; ++k)
                if (board[i + k][j + k] != player) {
                    win = false;
                    break;
                }
            if (win) return true;

            // 左下斜
            win = true;
            for (int k = 0; k < 5; ++k)
                if (board[i + 4 - k][j + k] != player) {
                    win = false;
                    break;
                }
            if (win) return true;
        }
    }
    return false;
}

// 回溯递归模拟棋局
void simulateGame(vector<vector<char>>& board, int step, bool whiteTurn) {
    if (step == N * N) { // 棋盘填满
        if (!hasFiveInARow(board, 'W') && !hasFiveInARow(board, 'B'))
            ++drawCount; // 平局
        return;
    }

    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            if (board[i][j] == '.') { // 空格可落子
                board[i][j] = whiteTurn ? 'W' : 'B';
                // 剪枝：若当前局面已分胜负，则跳过
                if (!hasFiveInARow(board, whiteTurn ? 'W' : 'B'))
                    simulateGame(board, step + 1, !whiteTurn);
                board[i][j] = '.'; // 回溯
            }
        }
    }
}

int main() {
    vector<vector<char>> board(N, vector<char>(N, '.')); // 初始化棋盘
    simulateGame(board, 0, true); // 白棋先手
    cout << drawCount << endl; // 输出平局总数
    return 0;
}

File: i am tired/blue/1.py
from itertools import product

# 棋盘大小
N = 5
# 棋盘格子数
TOTAL_CELLS = N * N

def is_draw(board):
    # 检查是否为平局（没有任何五子连珠的情况）
    def has_five_in_a_row(b, player):
        # 检查横、竖、斜方向是否有五子连珠
        for i in range(N):
            for j in range(N):
                if all(b[i][k] == player for k in range(j, min(j + 5, N))):  # 横向
                    return True
                if all(b[k][j] == player for k in range(i, min(i + 5, N))):  # 纵向
                    return True
        # 检查斜线
        for i in range(N - 4):
            for j in range(N - 4):
                if all(b[i + k][j + k] == player for k in range(5)):  # 右下斜
                    return True
                if all(b[i + 4 - k][j + k] == player for k in range(5)):  # 左下斜
                    return True
        return False

    # 将平铺的棋局转换为二维形式
    b = [board[i * N:(i + 1) * N] for i in range(N)]
    return not (has_five_in_a_row(b, 'W') or has_five_in_a_row(b, 'B'))

# 枚举所有棋局
draw_count = 0
for board in product('WB', repeat=TOTAL_CELLS):
    if is_draw(board):
        draw_count += 1

print(draw_count)
